### DISCUSSION TITLE
go test tsdb/example_test.go

### DISCUSSION THREAD

**Discussion by Tusenka** [prometheus/prometheus#15105] [Q&A]
go test tsdb/example_test.go                 ✘   main ●  22.02s  00:10:54 
# command-line-arguments [command-line-arguments.test]
tsdb/example_test.go:34:13: undefined: Open
tsdb/example_test.go:34:33: undefined: DefaultOptions
FAIL    command-line-arguments [build failed]
FAIL

**Comment by bboreham** [comment]
These errors arise because `example_test.go` uses items defined in the `tsdb` package but outside that file.
In general, Go code must be consumed as a package, not as a single file.

To run just that test, try:
```
go test -run Example ./tsdb
```
